### DISCUSSION TITLE
World Map settings, revamped launcher, better translation support...

### DISCUSSION THREAD

**Discussion by Tirlititi** [Albeoris/Memoria#957] [Releases]
![Release2024-11-17](https://github.com/user-attachments/assets/1bd87f86-2121-4eff-8eb5-edc66994bf14)
[COMPLETE CHANGELOG HERE](https://github.com/Albeoris/Memoria/wiki/Changelog-v2024.11.17)

## Features

- World map customization:
  - View distance (up to 4.5x), Mist/fog distance, FOV, camera distance and more
  - Look up! Use the right stick to look around, including up (can be disabled)
  - Alternative flight control (swaps sticks left and right vertical axis)
- Settings:
  - Auto-battle toggle [J/RT] can now be turned off
  - Invert camera vertical control
  - Invert flight control
  - New `Form party` button in the menu when `AllCharactersAvailable` is activated, accessible in battle (via menu access)
  - New cheat for Tetra master easy win
  - Thicker ATB bar (feat **Resinated**)
- Improved analog movement (closer to the PSX version)
- Improved the controls tutorial window to be more accurate with Memoria changes (thanks **faospark**)

## Launcher

- Full redesign with integrated mod manager
- A plethora of settings from `Memoria.ini` exposed, including some new settings
- Presets with three pre-made presets: Memoria default, Steam default and PSX default
- Button to save current settings as preset
- Button to copy the content of Memoria.log
- A changelog window will appear after an update or by clicking on Memoria's version.
- Button to auto-reorder mods
- Mods downloaded separately can now be installed via drag-and-drop
- Translations:
  - Spanish fixed by **Carú**
  - Brazilian-Portuguese fixed by **dkPsycho**
  - Chinese (Simplified) fixed by **imcloud**
  - Turkish written by **sangacanga1**

## Modding

- Translation support improvements (right-to-left, support of arabic texts and more)
- Improved modding system and capabilities
- Voice acting: Added `ForceMessageSpeed`, `ForceLanguage` and support for Special Events for Hot and Cold and Festival of the Hunt
- Sub-mods can have conditional values based on another mod being on/off
- Mods can include a preset (`Preset.ini`) applied when activated (with user prompt)

## Bugfixes

- Fixed some soft-locks
- Fixed manual trance broken with PSX GUI
- Fixed many gameplay issues
- Fixed music starting later in the track
- Fixed worldmap, fields and battle graphical issues
- Fixed some shader issues

@snouz @Tirlititi @DV666 @SamsamTS @faospark @barkermn01 @EricHu33 @Albeoris

<hr /><em>This discussion was created from the release <a href='https://github.com/Albeoris/Memoria/releases/tag/'>World Map settings, revamped launcher, better translation support...</a>.</em>

**Comment by Raulsalmeida** [comment]
Hi, my game doesn't launch after the update, do I have to reinstall everything?

**Reply by SamsamTS** [reply]
A hot fix is underway. Sorry for the inconvenience.

**Reply by Raulsalmeida** [reply]
Thanks!

**Reply by SamsamTS** [reply]
You might want to add that Settings.ini file back with this in it:
```
[Settings]
WindowMode =
ScreenResolution =
ActiveMonitor =
[Memoria]
IsX64 = True
IsDebugMode = False
CheckUpdates = True
```

**Comment by geordie0511** [comment]
Hi! Can't set resolution to 3840x2160 fullscreen (widescreen). Still see the borders left & right (used to work before the update). Display settings always reset to windows.

**Reply by SamsamTS** [reply]
Can you share your what's in your Settings.ini by any chance?

**Reply by geordie0511** [reply]
Where is the Settings.ini located ?

**Reply by SamsamTS** [reply]
It should be located in your game folder

**Reply by geordie0511** [reply]
Memoria.ini? there ist no Settings.ini

**Reply by SamsamTS** [reply]
Alright then. Create a new text file and rename it Settings.ini and put the following in it:
```
[Settings]
WindowMode =
ScreenResolution =
ActiveMonitor =
[Memoria]
IsX64 = True
IsDebugMode = False
CheckUpdates = True
```

**Reply by geordie0511** [reply]
This worked, thanks!

**Reply by charlie-in-spain** [reply]
Hi there, 

I've put those lines in Settings.ini but it just resets settings. When I launch the game it doesn't fit the whole screen of my monitor: 

![settingsini_memoria](https://github.com/user-attachments/assets/fe685f9d-c515-48f2-b04c-1f3206d4719c)

It does fit it if I choose the original screen instead, which is my laptop's. I've tried to put Settings.ini in the game folder and also in my mod's folder. 

I've checked the resolution of my monitor and it is the same as the one I choose in the launcher: 1920x1080 | 16:9

Thanks!

**Reply by SamsamTS** [reply]
@charlie-in-spain This will hopefully be fixed soon. In the mean time you can try to change the mode between `Fullscreen` and `Borderless Fullscreen`. If that still doesn't work you can also try to put a high resolution in the `Settings.ini` (the one in the game folder) something like `ScreenResolution = 10000x10000`

**Reply by charlie-in-spain** [reply]
Hi @SamsamTS 

I tried that but then the game's screen is a square that doesn't fit the whole screen of my monitor either.
If I put something like "3000x1000" it fits the length but not the height, really curious... 

I'll wait for that fix, anyway. Thanks!

**Comment by charlie-in-spain** [comment]
Hi again! 

I've noticed that if I select "normal" ATB mode in the launcher, the game will always play with "dynamic". It won't keep the option selected in the launcher. I can't find a parameter for this in memoria.ini file, to check if I can force this option to be set.

I don't know if this is also the reason why some wierd stuff happens in the game, like for example: 
-In some battles the command "Defend" is always shown instead of the different character's commands. For example, I want to choose "Steal" for Zidane, but the option automatically flips to "Defend" when selecting "Steal". 
-Freya appears in battle with Auto-float and Auto-reflect statuses from the beginning, and she is not carrying any equipment which provides them to her.
This seems to happen randomly, but the issue with ATB mode is always happening. 

I've tried to reinstall last Memoria patcher but it didn't fix it.

**Reply by SamsamTS** [reply]
I forgot about the ratio earlier, you'll have to keep 16:9 so 16000x9000 should do hopefully

The ATB mode is called Speed in the ini, it's in the Battle section. It can also be changed at any time in game via Config in the menu.

> In some battles the command "Defend" is always shown instead of the different character's commands. For example, I want to choose "Steal" for Zidane, but the option automatically flips to "Defend" when selecting "Steal".

That's a bug I'm trying to solve. Also the stick navigation is a bit too sensitive currently (will be fixed next version), you can try to increase the stick threshold in the Advanced tab of the launcher and see if that helps.

> Freya appears in battle with Auto-float and Auto-reflect statuses from the beginning, and she is not carrying any equipment which provides them to her.

No idea on that one, could be mod related.

**Reply by charlie-in-spain** [reply]
Hi! 

I've tried with 16000x9000 but it doesn't fit the screen either. In the launcher I'm not selecting any resolution: 

![launcher_resolution](https://github.com/user-attachments/assets/41306ba3-c572-4127-807c-de8cdc06ec5d)

I've also tried to select the first resolution as I did before: 1920x1080 | 16:9

Regarding ATB mode, I've selected "normal" via Config in the game's menu:

![atb_dynamic](https://github.com/user-attachments/assets/12d683eb-89f0-431c-b9ab-c906335cbbe0)

But it won't save this change when exiting and entering this menu again:

![atb_dynamic2](https://github.com/user-attachments/assets/a7eac6b8-692e-4457-9b7d-10bdc2f269c6)

Same happens in the launcher, when selecting ATB mode. You can select "Normal" but then in game "Dynamic" is automatically selected again.

**Reply by charlie-in-spain** [reply]
Hi!

Have you been able to reproduce the bug concerning ATB mode? I've also tried to edit the value in memoria.ini, but when I run the launcher ATB mode will always be in "Dynamic" by default:

![launcher_atbmode](https://github.com/user-attachments/assets/8fd4c76b-be6a-422f-9a5d-9494a85c76dd)

The issue is that also in game "Dynamic" is always set, no matter what mode you choose, as I showed you in the previous post.

According to memoria.ini, Speed can have these values: 
Speed (default 0) 0 = Default / 1 = Fast / 2 = Turn-based / 3 = Simultaneous, targets can't attack, turn-order respected / 4 = Simultaneous, targets can't attack / 5 = Simultaneous

I'd like to know the equivalence for the values in the launcher, since it only shows four: Normal / Fast / Turn-based / Dynamic. 
It looks like "Dynamic" could have 3 different modes according to memoria.ini (3 / 4 / 5), but in the launcher there's only "Dynamic". 

Thanks!

**Reply by SamsamTS** [reply]
Are there any other settings that do not save? Might be a write access issue with your `Memoria.ini` file. You could try deleting it (or just renaming it to keep a backup). A new one will be created when opening the launcher.

Another possibility is that a mod is forcing the ATB mode to dynamic. What mods are you using?

Dynamic in the launcher sets the Speed value to 5 (simultaneous) in the ini.

**Reply by charlie-in-spain** [reply]
Hi there, 

It was indeed the Memoria.ini file. I deleted it and then when I opened the launcher the ATB mode was working fine. Now I can select different ATB modes in the game and changes are saved. 

Thanks a lot!

**Reply by charlie-in-spain** [reply]
Hi! 

I've already tested the latest Memoria patch and now fullscreen works fine, adjusting the resolution to the whole screen. 
Thanks for the update, and great work!

**Reply by SamsamTS** [reply]
That's great to hear!

**Comment by Algarad** [comment]
Hi all,

I hope this is the correct thread to report this.

I went back to FF9 modding after a couple of month and found the cool revamped launcher!
I noticed a different behavior with how voice acting works now. For example, the dialogue between the two guards in Lindblum in the screenshot works different now.

![image](https://github.com/user-attachments/assets/ebe3cc09-aea3-493c-baa7-2ca1cc044690)

By reading memoria logs, I can see that now the game searches for _va_540_9_taken_0_, and not for _va_540_.
I assume that there is a new function for combined dialogues, something like an interaction within the dialogues between the two guards here.
I could not find anything about _multiplay-path_ or _taken_(num)_ voice files in the Wiki about it, but this is probably my fault.

> 01.12.2024 10:19:00 |M| , path:Voices/IT/276/va_540_taken_0, multiplay-path:Voices/IT/276/va_540_9_taken_0 (not found)
01.12.2024 10:19:01 |M| [VoiceActing] field:276, msg:541, text:Aureliano
01.12.2024 10:19:01 |M| Si può entrare a piedi al castello
01.12.2024 10:19:01 |M| sia da qui che dalla porta del drago
01.12.2024 10:19:01 |M| terrestre che sta sotto la “nebbia”.
>

I also noticed that If I talk with the guard again and again, the VA file value increments and changes from _va_540_9_taken_0_ to _va_540_9_taken_1_ and so on.

> 01.12.2024 10:19:09 |M| , path:Voices/IT/276/va_540_taken_1, multiplay-path:Voices/IT/276/va_540_9_taken_1 (not found)
01.12.2024 10:19:10 |M| [VoiceActing] field:276, msg:541, text:Aureliano
01.12.2024 10:19:10 |M| Si può entrare a piedi al castello
01.12.2024 10:19:10 |M| sia da qui che dalla porta del drago
01.12.2024 10:19:10 |M| terrestre che sta sotto la “nebbia”.
>

Let me know if I need to open a specific discussion thread, in case :)

Algarad

**Reply by SamsamTS** [reply]
Can you create an issue for it and copy this exact message?
I've looked into the code and there is indeed a problem. @barkermn01 did changes for the festival of the hunt and chocobo hot and cold but didn't take into account other languages nor kept it retro-compatible. I will at the very least ensure that it is retro compatible, sorry about that.

**Reply by Algarad** [reply]
Hi,

sure, will do!

**Comment by novacarolnathan** [comment]
I currently have AllCharactersAvailable in Memoria.ini set to 0 but the game still lets me switch them at any time, even to those who shouldn't be usable for story reasons (such as Eiko when she's kidnapped).

**Reply by Tirlititi** [reply]
If you're using the Playable Character Pack mod, it forces that ``AllCharactersAvailable`` option currently, because that's the only way to have them be in the party and I didn't want to have people wonder "when the extra character would show up" (they would never, if that option was de-activated).

You can however turn it off by editing the ``PlayableCharacterPack/Memoria.ini``.

If you don't use that mod, then it's something else, possibly a bug indeed.

**Reply by novacarolnathan** [reply]
I am not using that mod. I have Tantalus and Beatrix installed but not the Playable Character Pack specifically.

**Reply by SamsamTS** [reply]
Tantalus does have `AllCharactersAvailable = 1` as well

**Reply by novacarolnathan** [reply]
Ah thanks, that did it. Though is there a way to keep the ability to swap out the party from the menu but with any unavailable characters left out?

**Comment by moonstar207** [comment]
Hello, thanks as always for the updates!

So atm in battles seems the closeups to the characters when they're casting are disabled. Also when summoning, it shows as the eidolon instead of the attack name like "Shiva" instead of "Diamond Dust". Is this a setting I can turn off?

![ffix1](https://github.com/user-attachments/assets/6d47839d-9e6b-4ec3-8eaa-03ce33c59b43)

**Reply by SamsamTS** [reply]
Hello.
The lack character close up is likely due to `SFXRework`. You can turn it off by editing the `Memoria.ini` and changing it to `SFXRework = 0`
No idea about the eidolon names.

**Reply by moonstar207** [reply]
Oh yeah. I actually searched around and tried disabling that and it didn't change. Opened Memoria.ini again and turns out I didn't save the change lol. Thanks for that!

Yeah the eidolon name is weird, I don't think it was this way before?

**Reply by Tirlititi** [reply]
I still have "Diamond Light", "Terrestrial Rage", "G-force 199" etc.
I just checked and it must a mod doing that: apparently, the David Bowie mod and the Beatrix mod should be updated because they have that (unintentional) side-effect.

It might be that I'll fix the problem in Memoria itself instead. It's not clear to me whether mods should fix it or Memoria should fix it yet, I'll think about it. Thanks for pinpointing that problem.

**Reply by moonstar207** [reply]
Ohh, ok so it's the other mods doing that. Thanks for the insight!
